libdnf/conf/config_main.cpp

```diff
@@ -0,0 +1,885 @@
+    std::ostringstream res;
+    std::string::size_type start{0};
+    while (start < str_with_globs.length()) {
+        auto end = str_with_globs.find_first_of(" ,\n", start);
+        if (str_with_globs.compare(start, 5, "glob:") == 0) {
+            start += 5;
+            if (start >= str_with_globs.length()) {
+                break;
+            }
+            if (end == std::string::npos) {
+                add_from_files(res, str_with_globs.substr(start));
+                break;
+            }
+            if ((end - start) != 0) {
+                add_from_files(res, str_with_globs.substr(start, end - start));
+            }
+        } else {
+            if (end == std::string::npos) {
+                res << str_with_globs.substr(start);
+                break;
+            }
+            if ((end - start) != 0) {
+                res << str_with_globs.substr(start, end - start) << " ";
+            }
+        }
+        start = end + 1;
+    }
+    return res.str();
+}
+
+class ConfigMain::Impl {
+    friend class ConfigMain;
+
+    explicit Impl(Config & owner);
+
+    Config & owner;
+
+    OptionNumber<std::int32_t> debuglevel{2, 0, 10};
+    OptionNumber<std::int32_t> errorlevel{3, 0, 10};
+    OptionPath installroot{"/", false, true};
+    OptionPath config_file_path{CONF_FILENAME};
+    OptionBool plugins{true};
+    OptionStringList pluginpath{std::vector<std::string>{}};
+    OptionStringList pluginconfpath{std::vector<std::string>{}};
+    OptionPath persistdir{PERSISTDIR};
+    OptionBool transformdb{true};
+    OptionNumber<std::int32_t> recent{7, 0};
+    OptionBool reset_nice{true};
+    OptionPath system_cachedir{SYSTEM_CACHEDIR};
+    OptionBool cacheonly{false};
+    OptionBool keepcache{false};
+    OptionString logdir{"/var/log"};
+    OptionNumber<std::int32_t> log_size{1024 * 1024, str_to_bytes};
+    OptionNumber<std::int32_t> log_rotate{4, 0};
+    OptionStringList varsdir{VARS_DIRS};
+    OptionStringList reposdir{{"/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d"}};
+    OptionBool debug_solver{false};
+    OptionStringList installonlypkgs{INSTALLONLYPKGS};
+    OptionStringList group_package_types{GROUP_PACKAGE_TYPES};
+
+    OptionNumber<std::uint32_t> installonly_limit{3, 0, [](const std::string & value) -> std::uint32_t {
+                                                      if (value == "<off>") {
+                                                          return 0;
+                                                      }
+                                                      try {
+                                                          return static_cast<std::uint32_t>(std::stoul(value));
+                                                      } catch (...) {
+                                                          return 0;
+                                                      }
+                                                  }};
```

Comment:
The formatting looks not nice. Is it possible to improve it?

Reply:
@j-mracek I don't like it too. But this is result of clang-format. I thought about it. But I don't have good solution now.

Reply:
Maybe clang-format can be adjusted, I recall we wanted just four spaces indentation for these cases and it seemed possible, maybe @dmach will know.